File: 636.cpp
#include<iostream>
#include<string>
#include<sstream>

using namespace std;

class Solution {
public:
    vector<int> exclusiveTime(int n, vector<string>& logs) {
        vector<int> res(n,0);
        vector<pair<int,int>> t;
        int id,time;
        string state;
        stack<pair<int,int>> s;//pair id,start_time
        for(int i=0;i<logs.size();++i){
            istringstream in(logs[i]);
            in>>id;
            in>>state;
            if(state[1]=='s'){
                time=stoi(state.substr(7));
                state=state.substr(1,5);
            }
            else{
                time=stoi(state.substr(5));
                state=state.substr(1,3);
            }
            if(state=="start"){
                s.push({id,time});
            }
            else{
                int spent=time-s.top().second+1;
                res[id]+=spent;
                s.pop();
                if(!s.empty())
                    res[s.top().first]-=spent;
            }
            
        }
        return res;
        
    }
};


int main(){

}

File: Python3/636.py
class Solution:
    def exclusiveTime(self, n: int, logs: List[str]) -> List[int]:
        res = [0]*n
        stack = [] # (id, time)
        for log in logs:
            items = log.split(':')
            id, state, time = int(items[0]), items[1], int(items[2])
            if state == "start":
                stack.append((id, time))
            else:
                time_spent = time - stack[-1][1] + 1
                res[id] += time_spent
                stack.pop()
                if stack:
                    prev_id = stack[-1][0]
                    res[prev_id] -= time_spent
        return res
